hathor/indexes/rocksdb_tokens_index.py

```diff
@@ -0,0 +1,485 @@
+        self.log.debug('remove transaction', token=token_uid.hex(), tx=tx_hash.hex(), timestamp=timestamp)
+        self._db.delete((self._cf, self._to_key_txs(token_uid, _TxIndex(tx_hash, timestamp))))
+
+    def _add_mint_utxo(self, token_uid: bytes, tx_hash: bytes, index: int) -> None:
+        self.log.debug('add mint utxo', token=token_uid.hex(), tx=tx_hash.hex(), index=index)
+        self._db.put((self._cf, self._to_key_mint(token_uid, TokenUtxoInfo(tx_hash, index))), b'')
+
+    def _add_melt_utxo(self, token_uid: bytes, tx_hash: bytes, index: int) -> None:
+        self.log.debug('add melt utxo', token=token_uid.hex(), tx=tx_hash.hex(), index=index)
+        self._db.put((self._cf, self._to_key_melt(token_uid, TokenUtxoInfo(tx_hash, index))), b'')
+
+    def _remove_mint_utxo(self, token_uid: bytes, tx_hash: bytes, index: int) -> None:
+        self.log.debug('remove mint utxo', token=token_uid.hex(), tx=tx_hash.hex(), index=index)
+        self._db.delete((self._cf, self._to_key_mint(token_uid, TokenUtxoInfo(tx_hash, index))))
+
+    def _remove_melt_utxo(self, token_uid: bytes, tx_hash: bytes, index: int) -> None:
+        self.log.debug('remove melt utxo', token=token_uid.hex(), tx=tx_hash.hex(), index=index)
+        self._db.delete((self._cf, self._to_key_melt(token_uid, TokenUtxoInfo(tx_hash, index))))
+
+    def _create_genesis_info(self) -> None:
+        self._create_token_info(
+            settings.HATHOR_TOKEN_UID,
+            settings.HATHOR_TOKEN_NAME,
+            settings.HATHOR_TOKEN_SYMBOL,
+            settings.GENESIS_TOKENS,
+        )
+
+    def _add_to_total(self, token_uid: bytes, amount: int) -> None:
+        key_info = self._to_key_info(token_uid)
+        old_value_info = self._db.get((self._cf, key_info))
+        if token_uid == settings.HATHOR_TOKEN_UID and old_value_info is None:
+            self._create_genesis_info()
+            old_value_info = self._db.get((self._cf, key_info))
+        assert old_value_info is not None
+        dict_info = self._from_value_info(old_value_info)
+        dict_info['total'] += amount
+        new_value_info = self._to_value_info(dict_info)
+        self._db.put((self._cf, key_info), new_value_info)
+
+    def _subtract_from_total(self, token_uid: bytes, amount: int) -> None:
+        key_info = self._to_key_info(token_uid)
+        old_value_info = self._db.get((self._cf, key_info))
+        if token_uid == settings.HATHOR_TOKEN_UID and old_value_info is None:
+            self._create_genesis_info()
+            old_value_info = self._db.get((self._cf, key_info))
+        assert old_value_info is not None
+        dict_info = self._from_value_info(old_value_info)
+        dict_info['total'] -= amount
+        new_value_info = self._to_value_info(dict_info)
+        self._db.put((self._cf, key_info), new_value_info)
+
+    def _add_authority_utxo(self, tx: BaseTransaction, index: int) -> None:
+        """ Add tx to mint/melt indexes and total amount
+        """
```

Comment:
The function name seems misleading. It handles non-authority utxos as well.

Reply:
Renamed to `_add_utxo`.